spinoso-string/src/ascii_string/ascii_string.rs

```diff
@@ -0,0 +1,516 @@
+use alloc::vec::Vec;
+use core::fmt;
+use core::ops::Range;
+use core::slice::SliceIndex;
+
+use bstr::{BStr, ByteSlice, ByteVec};
+
+use crate::codepoints::InvalidCodepointError;
+use crate::encoding::Encoding;
+use crate::iter::{Bytes, IntoIter, Iter, IterMut};
+use crate::ord::OrdError;
+
+#[derive(Default, Clone)]
+pub struct AsciiString {
+    inner: Vec<u8>,
+}
+
+// Constructors
+impl AsciiString {
+    pub fn new(buf: Vec<u8>) -> Self {
+        Self { inner: buf }
+    }
+}
+
+impl fmt::Debug for AsciiString {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Utf8String")
+            .field("buf", &self.as_bstr())
+            .field("encoding", &Encoding::Ascii)
+            .finish()
+    }
+}
+
+// Debug
+impl AsciiString {
+    #[inline]
+    #[must_use]
+    pub fn as_bstr(&self) -> &BStr {
+        self.inner.as_bstr()
+    }
```

Comment:
Good call. Since removed!

I just recently realised that the `fmt::Debug` on the encoded implementations is likely to never be called (apart from maybe in testing etc). I'm curious, do you think the encoded implementations should be part of the public API? I can't currently forsee any real use case TBH.